jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Components.java

```diff
@@ -43,4 +44,6 @@ public interface Components
      * @return A Map, which may be an empty map that discards all items.
      */
     Attributes getCache();
+
+    ComplianceViolation.Listener getComplianceViolationListener();
```

Comment:
I don't like exposing `ComplianceViolation.Listener` into a method signature.
I'm fine using it as a bean, or as an attribute, but the mechanism of configuring `ComplianceViolation.Listener`s already feels fragile enough that exposing it further does not seem a good idea.

Furthermore, `Components` is supposed to be used by users, to access long-lived components that they can use in their applications to schedule tasks, or dispatch to thread pool, etc.

By allowing them to call `getComplianceViolationListener()` is not good because they won't be able to do anything with it -- it's an implementation detail that we need internally to call this method to retrieve the listener, so it should not be present in a user-facing class like `Components`.

I much prefer using request attributes to retrieve the listener.